plugins/by-name/snacks/default.nix

```
@@ -0,0 +1,259 @@
+{ lib, ... }:
+let
+  inherit (lib) types;
+  inherit (lib.nixvim) defaultNullOpts;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "snacks";
+  originalName = "snacks.nvim";
+  package = "snacks-nvim";
+
+  maintainers = [ lib.maintainers.HeitorAugustoLN ];
+
+  settingsOptions = {
+    bigfile = {
+      enabled = defaultNullOpts.mkBool true ''
+        Whether to enable `bigfile` plugin. 
+      '';
+
+      notify = defaultNullOpts.mkBool true ''
+        Whether to show notification when big file detected.
+      '';
+
+      size = defaultNullOpts.mkLua "1.5 * 1024 * 1024" ''
+        The size at which a file is considered big.
+      '';
+
+      setup =
+        defaultNullOpts.mkLua
```

```suggestion
        defaultNullOpts.mkRaw
```
We have recently decided to discourage the use of `mkLua` because it adds a silent automatic conversion to `rawLua`. We now prefer to use `defaultNullOpts.mkRaw` to force the user to explicitly use a `__raw` value.